ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -102,11 +117,16 @@
           @Override
           public void run() {
             try {
-              HelixManager manager =
-                  helixFactory.getZKHelixManager(clusterName, instanceName, InstanceType.SPECTATOR, zkConnectStr);
-              logger.info("Connecting to Helix manager at {}", zkConnectStr);
-              manager.connect();
-              logger.info("Established connection to Helix manager at {}", zkConnectStr);
+              HelixManager manager;
+              if (!dcName.equals(clusterMapConfig.clusterMapDatacenterName)) {
```

Comment:
skip the HelixManager for local DC. Not sure how much delay it will bring. (Basically we change 4 concurrent threads to 1 thread first and 3 concurrent threads)

Reply:
shouldn't be a problem, though is there a reason to not initialize overrides in parallel?

Reply:
for race condition concerns, will talk to you offline.